### PR TITLE
cmip_urls to get file paths

### PR DESCRIPTION
**Pull request by mdsumner**
as mentioned in #9 

I've got concerns about 

- [x] the https vs http thing (it seems to handle finding a mix of these)
- [x] the speed, it's quite slow and maybe can be sped up see #11 
- [x] the way I collapse out and string-sub the urls is pretty crass and possibly flaky
- [ ] (probably) I should not slam the timeout option, but check if it's already set

This obviously overlaps with the internal function `cmip_download_one()` and this could be refactored to a common basis.

### CODE REVIEW

**Comment by mdsumner** [comment]
occurred to me that this could simply be included in the "results" object, I was treating that thing blindly like a foreign object but I realize now it's a data.table 

if you think it should just be a column on the results from a search I'll rewrite it do that

File: R/cmip-urls.R [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    if (httr::http_error(info)) {
+      next;
+    }
+    info <- try(httr::content(info)$response$docs)
+    if (inherits(info, "try-error")) {
+      next;
+    }
+    ## FIXME/MDSumner bit worried here about what the structure could be in here (can there be multiple?)
+    tx <- grep("HTTPServer", unlist(info[[1]]$url), value = TRUE)
```
Comment (eliocamp) [inline R/cmip-urls.R:40]: Is this the equivalent to this line in cmip_download_one? 

https://github.com/eliocamp/rcmip6/blob/2ebd0d95d15caa9fec39534587add39cb368182a/R/cmip-download.R#L77

Is this code more robust?

Comment (mdsumner) [reply R/cmip-urls.R:40]: oh yes i see, actually I got burned by my PR code yesterday by data at nci.org - 😅 will replace with yours 👌

Comment (mdsumner) [reply R/cmip-urls.R]: I updated

**Comment by eliocamp** [comment]
Sorry it took so long for me to merge this!

**Pull request merged**